src/fheroes2/dialog/dialog_selectfile.cpp

```diff
@@ -461,6 +608,14 @@ namespace
                     result = System::concatPath( Game::GetSaveDir(), filename + Game::GetSaveFileExtension() );
                 }
             }
+            else if ( le.MouseClickLeft( nameHeaderRoi ) && settings.GetSaveFileSortingMethod() != SaveFileSortingMethod::FILENAME ) {
+                listUpdated = true;
+                needRedraw = switchFileSorting( settings.GetSaveFileSortingMethod() == SaveFileSortingMethod::FILENAME );
+            }
+            else if ( le.MouseClickLeft( dateHeaderRoi ) && settings.GetSaveFileSortingMethod() != SaveFileSortingMethod::TIMESTAMP ) {
+                listUpdated = true;
+                needRedraw = switchFileSorting( settings.GetSaveFileSortingMethod() == SaveFileSortingMethod::FILENAME );
```

Comment:
The same as previous comment, but `true` should be passed:
```suggestion
                needRedraw = switchFileSorting( true );
```